scheduler/src/cook/mesos/api.clj

```diff
@@ -960,7 +960,7 @@
   [db entity]
   (if entity
     (compute-cluster-entity->map entity)
-    (->> (cc/cluster-name-hack)  ; Get the default cluster.
+    (->> (cc/get-default-cluster-name-for-legacy)  ; Get the default cluster.
```

Comment:
Can we just store the entire compute cluster in the variable instead of going out to the DB?

Reply:
Once the patch rolls out, after a week, we'll need to go out to the db every time (and it'll be cached, so that will be cheap), therefore this only optimizes the slow-case.

We still need to map the ComputeCluster to its corresponding datomic entity, because thats what this code expects. Do you want me to add a simple cache here? We can't cache the datomic entity, but we can cache the dictionary generated by compute-cluster-entity->map, and return it every time. Unless you see a nicer way, I don't think that that optimization is worth adding the extra atom and 2 lines of code. 
(when-not @cached-default-compute-cluster-for-legacy-as-a-response-map
    (reset! @cached.... (... the same code as above...))

Reply:
If we need the entity, why not just have `get-default-cluster-for-legacy` that returns the entity (cached on startup) and then read the name out of that if needed? It lets us avoid all these gymnastics to get back to the entity.

Reply:
Isn't there a problem when you keep a datomic entity cached in memory; it keeps a lot of extra datomic state cached alive in memory. Thats why we don't do that anymore in Cook?

Reply:
It could just be a map with the contents of the compute cluster instead of the entity itself to avoid that problem